app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list) -> None:
+    person_list = []
+    for person in people:
+        person_instance = Person(name=person["name"], age=person["age"])
+        person_list.append(person_instance)
+
+    for person in people:
+        person_instance = Person.people[person["name"]]
+        if "wife" in person and person["wife"] is not None:
+            person_instance.wife = Person.people[person["wife"]]
+        elif "husband" in person and person["husband"] is not None:
+            person_instance.husband = Person.people[person["husband"]]
```

Comment:
The code assumes that the spouse (wife/husband) is already created and present in the `Person.people` dictionary. Ensure that the input list `people` is ordered or structured such that all referenced spouses are created before they are assigned.